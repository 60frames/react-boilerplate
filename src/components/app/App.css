.root {
    position: absolute;
    top: 50%;
    right: 50%;
    transform: translate(50%, -50%);
    width: 90%;
    max-width: 500px;
}

.logo {
    background-image: url('components/app/Logo.svg');
    background-repeat: no-repeat;
    background-position: 50% 50%;
    background-size: 100%;
    position: relative;
    width: 90%;
    max-width: 400px;
    text-indent: -9999em;
    overflow: hidden;
    margin: 0 auto 2em;
    animation-duration: 3s;
    animation-iteration-count: infinite;
    animation-name: breathe;
}

.logo:before {
    display: block;
    content: '';
    width: 100%;
    padding-top: 32.564102564103%;
}

.logoContent {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    cursor: pointer;
}

@keyframes breathe {
    0% {
        transform: translate(0%, 0%);
    }
    50% {
        transform: translate(0%, 8%);
    }
    100% {
        transform: translate(0%, 0%);
    }
}
