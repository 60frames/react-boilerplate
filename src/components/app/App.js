import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { Link } from 'react-router';
import styles from 'components/app/App.css';

const DEFAULT_TITLE = '60fram.es React Boilerplate';

function App({ children }) {
  return (
    <div className={styles.root}>
      <Helmet
        titleTemplate={`%s | ${DEFAULT_TITLE}`}
        defaultTitle={DEFAULT_TITLE}
      />
      <div className={styles.logo}>
        <Link to="/" title="Home" className={styles.logoContent}>
          60fram.es
        </Link>
      </div>
      {children}
    </div>
  );
}

App.propTypes = {
  children: PropTypes.node
};

export default App;
