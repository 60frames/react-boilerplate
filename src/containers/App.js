import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import { fetchPostsIfNeeded } from '../actions';

const DEFAULT_TITLE = 'React Boilerplate';

class App extends Component {

    componentDidMount() {
        const { dispatch } = this.props;
        App.fetchData({ dispatch });
    }

    render() {
        return (
            <div>
                <Helmet titleTemplate={`%s | ${DEFAULT_TITLE}`} defaultTitle={DEFAULT_TITLE} />
                {this.props.children}
                {JSON.stringify(this.props.data)}
            </div>
        );
    }
}

App.fetchData = function({ dispatch }) {
    return dispatch(fetchPostsIfNeeded());
};

function mapStateToProps(state) {
    return {
        ...state.posts
    };
}

export default connect(mapStateToProps)(App);
