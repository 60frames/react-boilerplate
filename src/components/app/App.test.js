/* eslint-env jest */

jest.mock('react-helmet', () => 'Helmet');
jest.mock('react-router');

import React from 'react';
import renderer from 'react-test-renderer';
import App from 'components/app/App';

describe('components/app/App', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <App>
        Hello world.
      </App>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
