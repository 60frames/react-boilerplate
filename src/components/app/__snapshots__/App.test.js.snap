// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/app/App renders correctly 1`] = `
<div
  className="root"
>
  <Helmet
    defaultTitle="60fram.es React Boilerplate"
    titleTemplate="%s | 60fram.es React Boilerplate"
  />
  <div
    className="logo"
  >
    <Link
      className="logoContent"
      title="Home"
      to="/"
    >
      60fram.es
    </Link>
  </div>
  Hello world.
</div>
`;
